synapse/lib/stormlib/model.py

```diff
@@ -812,3 +814,189 @@ async def _methCopyTags(self, src, dst, overwrite=False):
+            rgx = s_infotech.cpe22_regex.match(v2_2)
+            if rgx is not None and rgx.group() == v2_2:
+                valu22 = v2_2
+
+        async with self.runt.snap.getNodeEditor(n) as proto:
+
+            # If both values are populated, this node is valid
+            if valu23 is not None and valu22 is not None and not force:
+                if self.runt.debug:
+                    mesg = f'DBG: itSecCpeFixup({reprvalu}): Node is valid, no migration necessary.'
+                    await self.runt.printf(mesg)
+
+                await proto.setData('migration.s.itSecCpeFixup', {
+                    'status': 'success',
+                })
+
+                return True
+
+            if valu23 is None and valu22 is None:
+                invalid = 'Primary property and :v2_2 are both invalid.'
+
+            if invalid:
+                # Invalid 2.3 string and no/invalid v2_2 prop. Nothing
+                # we can do here so log, mark, and go around.
+                mesg = f'itSecCpeFixup({reprvalu}): Unable to migrate due to invalid data. {invalid}'
+                await self.runt.warn(mesg)
+
+                await proto.setData('migration.s.itSecCpeFixup', {
+                    'status': 'failed',
+                    'reason': invalid,
+                })
+
+                return False
+
+            if prefer_v22:
+                valu = valu22 or valu23
+            else:
+                valu = valu23 or valu22
+
+            # Re-normalize the data from the 2.3 or 2.2 string, whichever was valid.
+            norm, info = modl.norm(valu)
+            subs = info.get('subs')
+
+            edits = []
+            nodedata = {}
+
+            if norm != curv:
+                # The re-normed value is not the same as the current value.
+                # Since we can't change the primary property, store the
+                # updated value in nodedata.
+                if self.runt.debug:
+                    mesg = f'DBG: itSecCpeFixup({reprvalu}): Stored updated primary property value to nodedata: {curv} -> {norm}.'
+                    await self.runt.printf(mesg)
+
+                nodedata['valu'] = norm
+
+            # Iterate over the existing properties
+            for propname, propcurv in n.props.items():
+                subscurv = subs.get(propname)
+                if subscurv is None:
+                    continue
+
+                if propname == 'v2_2' and isinstance(subscurv, list):
+                    subscurv = s_infotech.zipCpe22(subscurv)
+
+                # Values are the same, go around
+                if propcurv == subscurv:
+                    continue
+
+                nodedata.setdefault('updated', [])
+                nodedata['updated'].append(propname)
+
+                # Update the existing property with the re-normalized property value.
+                await proto.set(propname, subscurv, ignore_ro=True)
+
+            nodedata['status'] = 'success'
```

Comment:
could just set status to success when initializing the dict since we can't fail down here